### PR TITLE
Add Excel/ODS processing, ignore null bytes

### LINKED ISSUE #51: Bulk: Cope with win-1252 files
The code assumes Unicode and barfs if a Win-1252 file is uploaded.
This seems plausible from Windows systems (it happened today), and we should at least try and parse it as that too before giving up.

### LINKED ISSUE #52: Bulk: Cope with extraneous commas/columns/null bytes
If a row of the uploaded CSV has more columns than expected, it errors.

### LINKED ISSUE #56: Bulk: Allow Excel file upload

### PR DESCRIPTION
**Pull request by dracos**
Fixes #56. Fixes #52. Fixes #51.

### CODE REVIEW

File: bulk_lookup/management/commands/process_bulk_lookups.py [hunk 1]
```diff
@@ -19,6 +21,8 @@
 from mapit.models import Postcode, Area, Generation
 from mapit.views.areas import add_codes
 
+defusedxml.defuse_stdlib()
```
Comment (mhl) [inline bulk_lookup/management/commands/process_bulk_lookups.py:24]: Interesting - I didn't know about this!

File: mapit_mysociety_org/templates/pricing.html [hunk 2]
```diff
@@ -68,7 +68,7 @@ <h2 class="pricing__option__cost">£{{ PRICING.2.price }}/mth</h2>
                 {% else %}Max {{ PRICING.2.calls }} calls per month
                 {% endif %}
             </li>
-            <li><strong>Free</strong> use of our <a href="{% url "home" %}">CSV data mapper</a></li>
+            <li><strong>Free</strong> use of our <a href="{% url "home" %}">data mapper</a></li>
```
Comment (mhl) [inline mapit_mysociety_org/templates/pricing.html:71]: I'm guessing the changes to copy in this file should be in another commit, or another PR?

File: bulk_lookup/csv.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    won't work. https://github.com/pyexcel/pyexcel-io/issues/34 )"""
+    def __getattribute__(self, name):
+        if name == 'read':
+            raise AttributeError
+        return super(mmmap, self).__getattribute__(name)
+
+
+class PyExcelReader(object):
+    # Set up an iterator for reading in a CSV/XLS/XLSX/ODS file
+    def __init__(self, file):
```
Comment (mhl) [inline bulk_lookup/csv.py]: I'd probably rename this parameter `file_field`, since I assumed this was a normal file-like object and then got puzzled about where the `.open()` method came from. Also then it's not shadowing the `file` constructor.

**Review by mhl — COMMENTED** [review]
This looks fine to me (I just left a small parameter naming comment inline) although the pyexcel API confused me enough that it's hard to be sure about those bits.

I think that would make me worry about making changes to this code, though, and I would certainly feel more comfortable if I had to do that in the future if there were some tests that used the PyExcelReader class to take some minimal Excel / ODS / CSV files committed in the repo and parsed them.

Anyway, looks good :+1:

File: mapit_mysociety_org/templates/pricing.html (hunk 2 above)
Comment (dracos) [reply mapit_mysociety_org/templates/pricing.html:71]: It now handles Excel files, so removing "CSV" here made sense?

File: bulk_lookup/csv.py (hunk 3 above)
Comment (dracos) [reply bulk_lookup/csv.py]: Yep, will do.

**Comment by dracos** [comment]
Yeah, I found the API hard to follow too, though it works well once you do (think it might be a bit over-engineered with respect to plugins etc, but it did enough that it did feel easier than factoring out all the relevant bits, meh). There are some CSV tests in tests.py (using StringIO), but I'll add some example Excel/ODS files, yep.

File: mapit_mysociety_org/templates/pricing.html (hunk 2 above)
Comment (mhl) [reply mapit_mysociety_org/templates/pricing.html:71]: Oh yeah, sorry...
